src/app/gateways/views/BuiltinGatewayDetailView.vue

```diff
@@ -18,29 +18,19 @@
       }"
     >
       <AppView>
-        <DataSource
+        <DataLoader
```

Comment:
If I follow what you are doing here then this should be a DataSource which is non-blocking, the idea is you can then pass its data and errors into the single DataLoader (the DataLoader _is_ blocking).

Also, and perhaps separately to this PR, I've noticed that this DataSource should probably be in the Tabs view and its data passed down seeing as we use it in both tabs. Let me know if this point isn't clear

Reply:
This needs to be blocking because the result of it becomes part of the next DataLouder’s `props.src`: the request URL depends on data from this response.

Reply:
Oh I see, 1 sec lemme look again

Reply:
I think if you use:

```
 :src="typeof data === 'undefined' ? '' : `/meshes/${route.params.mesh}/dataplanes/for/${data.selectors[0].match['kuma.io/service']}?page=${route.params.page}&size=${route.params.size}&search=${route.params.s}`"

```

For the `:src` for the DataLoader and remove its `v-if="data"` (and remember to turn the wrapping one into a DataSource) then we will get what we want here. lmk

Reply:
But why wouldn’t I just use DataLoader on the top level if I want it to be blocking anyway?

Reply:
🤔 actually yeah I guess its fine as it is also, I suppose I'm just automatically thinking of the "most of the time we should only have one DataLoader thing" - also I think if we move the MeshGateway call to the route above this one to share it, you'll have to do the sort of thing in the suggestion eventually anyway. But yeah, this is fine I think. Will resolve this one.